helix-term/src/handlers/completion.rs

```diff
@@ -291,6 +302,197 @@ fn request_completion(
     });
 }
 
+fn path_completion(
+    cursor: usize,
+    text: helix_core::Rope,
+    doc: &helix_view::Document,
+) -> Option<BoxFuture<'static, anyhow::Result<Vec<CompletionItem>>>> {
+    if !doc.supports_path_completion() {
+        return None;
+    }
+
+    // TODO find a good regex for most use cases (especially Windows, which is not yet covered...)
+    // currently only one path match per line is possible in unix
+    static PATH_REGEX: Lazy<rope::Regex> =
+        Lazy::new(|| rope::Regex::new(r"((?:~|\$HOME|\$\{HOME\})?(?:\.{0,2}/)+.*)$").unwrap());
+
+    let cur_line = text.char_to_line(cursor);
+    let start = text.line_to_char(cur_line).max(cursor.saturating_sub(1000));
+    let line_until_cursor = text.slice(start..cursor).regex_input_at(..);
+
+    let (dir_path, typed_file_name) = PATH_REGEX.search(line_until_cursor).and_then(|m| {
+        let matched_path = &text.byte_slice(m.start()..m.end()).to_string();
```

Comment:
Now that we `slice` and then use `regex_input_at` we don't have the byte indices automatically adjusted for us. (They're relative to the slice `text.slice(start..cursor)` - `regex_input_at` made them absolute to `text` previously.)

To adjust this we'll need to grab the start byte (probably easiest to do `let start_byte = text.char_to_byte(start);` after `start` is bound) and then our matched path would be `text.byte_slice(start_byte + m.start()..start_byte + m.end())`

Reply:
Whoops, I guess I wasn't thorough enough... thanks.